airbyte-integrations/connectors/source-intercom/source_intercom/source.py

```diff
@@ -105,6 +105,10 @@ def filter_by_state(self, stream_state: Mapping[str, Any] = None, record: Mappin
     def parse_response(self, response: requests.Response, stream_state: Mapping[str, Any], **kwargs) -> Iterable[Mapping]:
         record = super().parse_response(response, stream_state, **kwargs)
 
+        # set initial state to use start_date from spec
+        if not stream_state:
```

Comment:
1. It checks if state for a resource eg. `conversations` is not present and in such cases uses `start_date`
2. Added additional check just in case empty object or something is passed for state initially
3. Added